pkg/cache/context.go

```diff
@@ -1053,25 +1014,28 @@ func (ctx *Context) getApplication(appID string) *Application {
 
 func (ctx *Context) RemoveApplication(appID string) error {
```

Comment:
This method is used by testing only. Could we use `RemoveApplicationInternal` to replace it? It seems we can make `RemoveApplicationInternal` return `boolean` to indicate successful remove for testing.

Reply:
It needs a cleanup, might be better to do that via a seperate jira. That makes it clear we're cleaning up dead code.  We should keep the RemoveApplication name and remove the RemoveApplicationInternal name. It is not really internal as it gets called from outside the context.

Reply:
filed: [YUNIKORN-2782](https://issues.apache.org/jira/browse/YUNIKORN-2782)